pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -1463,13 +1526,56 @@ func doActivate(ctx *domainContext, config types.DomainConfig,
 			// do nothing
 		case zconfig.Format_CONTAINER:
 			snapshotID := containerd.GetSnapshotID(ds.FileLocation)
-			if err := ctx.casClient.MountSnapshot(snapshotID, cas.GetRoofFsPath(ds.FileLocation)); err != nil {
+			rootPath := cas.GetRoofFsPath(ds.FileLocation)
+			if err := ctx.casClient.MountSnapshot(snapshotID, rootPath); err != nil {
 				err := fmt.Errorf("doActivate: Failed mount snapshot: %s for %s. Error %s",
 					snapshotID, config.UUIDandVersion.UUID, err)
 				log.Error(err.Error())
 				status.SetErrorNow(err.Error())
 				return
 			}
+
+			log.Warnf("Config: \n%+v\n", config)
+			log.Warnf("Status: \n%+v\n", *status)
+
+			metadataPath := filepath.Join(rootPath, "meta-data")
+
+			curCIVersion, err := getVersionFromMetaFile(metadataPath)
```

Comment:
How does this use of meta-data relate to the meta-data content created by EVE/domainmgr and placed in the iso image?

Reply:
in the case of ISO image we are using the original cloud-init daemon with the data source [NoCloud](https://cloudinit.readthedocs.io/en/latest/reference/datasources/nocloud.html#). it expects a `meta-data` file filled with metadata, like you would find it in the [EC2 metadata service](https://cloudinit.readthedocs.io/en/latest/reference/datasources/ec2.html#amazon-ec2). here we can define a whole number of metadata fields:
```
ami-id
ami-launch-index
ami-manifest-path
block-device-mapping/
hostname
instance-id
instance-type
local-hostname
local-ipv4
placement/
public-hostname
public-ipv4
public-keys/
reservation-id
security-groups
```
Of those fields the `meta-data` file on our VMs only use `instance-id` and `local-hostname` of which the `instance-id` is especially important, since it contains the version of the cloud-init config.

In the case of container as edge-app, we don't rely on a cloud-init daemon, but parse the config manually. Thus of all metadata fields, only cloud-init version remains necessary - this is why I decided to include only the field `instance-id` in its original format in the `meta-data` file.

Reply:
OK